core/scripts/ocr2vrf/util.go

```diff
@@ -520,3 +532,167 @@ func requestRandomnessCallbackBatch(
 
 	return requestID
 }
+
+func getDKGLatestConfigDetails(e helpers.Environment, dkgAddress string) dkgContract.LatestConfigDetails {
+	dkg := newDKG(common.HexToAddress(dkgAddress), e.Ec)
+	dkgConfig, err := dkg.LatestConfigDetails(nil)
+	helpers.PanicErr(err)
+
+	return dkgConfig
+}
+
+func getVRFLatestConfigDetails(e helpers.Environment, beaconAddress string) vrf_beacon.LatestConfigDetails {
+	beacon := newVRFBeacon(common.HexToAddress(beaconAddress), e.Ec)
+	beaconConfig, err := beacon.LatestConfigDetails(nil)
+	helpers.PanicErr(err)
+
+	return beaconConfig
+}
+
+func getDKGKeyData(e helpers.Environment, dkgAddress string, keyID, configDigest [32]byte) dkgContract.KeyDataStructKeyData {
+	dkg := newDKG(common.HexToAddress(dkgAddress), e.Ec)
+	keyData, err := dkg.GetKey(nil, keyID, configDigest)
+	helpers.PanicErr(err)
+
+	return keyData
+}
+
+func getKeyID(e helpers.Environment, beaconAddress string) [32]byte {
+	beacon := newVRFBeacon(common.HexToAddress(beaconAddress), e.Ec)
+	keyID, err := beacon.SKeyID(nil)
+	helpers.PanicErr(err)
+	return keyID
+}
+
+func getPublicKey(e helpers.Environment, dkgAddress string, keyID, configDigest [32]byte) kyber.Point {
+	keyData := getDKGKeyData(e, dkgAddress, keyID, configDigest)
+	kg := &altbn_128.G2{}
+	pk := kg.Point()
+	err := pk.UnmarshalBinary(keyData.PublicKey)
+	helpers.PanicErr(err)
+	return pk
+}
+
+func getHashToCurveMessage(e helpers.Environment, height uint64, confDelay uint32, vrfConfigDigest [32]byte, pk kyber.Point) *altbn_128.HashProof {
+	blockNumber := big.NewInt(0).SetUint64(height)
+	block, err := e.Ec.BlockByNumber(context.Background(), blockNumber)
+	helpers.PanicErr(err)
+	b := ocr2vrftypes.Block{
+		Height:            height,
+		ConfirmationDelay: confDelay,
+		Hash:              block.Hash(),
+	}
+	h := b.VRFHash(vrfConfigDigest, pk)
+	return altbn_128.NewHashProof(h)
+}
+
+func getVRFSignature(e helpers.Environment, beaconAddress string, height, confDelay uint64) (proofG1X, proofG1Y *big.Int) {
+	// get transmission logs from requested block to latest block
+	// TODO: index transmission logs by height and confirmation delay to
+	// make the FilterQuery call more efficient
+	latestBlock, err := e.Ec.HeaderByNumber(context.Background(), nil)
+	helpers.PanicErr(err)
+	query := ethereum.FilterQuery{
+		FromBlock: big.NewInt(0).SetUint64(height),
+		ToBlock:   latestBlock.Number,
```

Comment:
In my testing I set this to `height + 200` or something, we usually get a response quite quickly. Narrows down the search space by a lot especially if it's an older proof.

Reply:
makes sense. Added a new CLI parameter for `searchWindow` and set default to 200